### DISCUSSION TITLE
Tree Grid Layout

### DISCUSSION THREAD

**Discussion by attran1** [emilk/egui#5520] [Q&A]
hello all,

I would like to write an UI to show data in a tree grid format something like this link https://www.htmlelements.com/docs/grid-tree-grid/.  I looked in Egui/grid and egui_extras/table.  I don't see anyway where I can collapse the parent tree.  I could not find any sample codes that have this capability.  Does anyone have experience in this area?  Please give me an advice on where to start.  I am new to Rust.

Thank you,
Anthony

**Comment by attran1** [comment]
Hello all,

I have not hear from anyone on this thread.  Can someone tell me if there are any crate that can provide this functionality?

Regards,
Anthony

**Comment by YgorSouza** [comment]
The list of existing crates is on <https://github.com/emilk/egui/wiki/3rd-party-egui-crates>. I don't know if there is any that has this tree +  table functionality. [`egui_table`](https://github.com/rerun-io/egui_table) might be close, since it has collapsible rows, but I don't think it is quite the same thing.
